### PR TITLE
Prevent duplicate actor identifiers, refs #13373

### PR DESCRIPTION
**Pull request by mcantelon**
* Added setting to enable prevention of duplicate authority record identifiers
* If setting is enabled: prevent entry of duplicates via the web GUI
* If setting is enabled: prevent import of duplicates via CSV and XML import

### CODE REVIEW

File: apps/qubit/modules/settings/actions/identifierAction.class.php [hunk 1]
```diff
@@ -30,7 +30,8 @@ class SettingsIdentifierAction extends DefaultEditAction
       'identifier_counter',
       'separator_character',
       'inherit_code_informationobject',
-      'inherit_code_dc_xml');
+      'inherit_code_dc_xml',
+      'prevent_duplicate_actor_identifiers');
```
Comment (djjuhasz) [inline apps/qubit/modules/settings/actions/identifierAction.class.php]: Cosmetic: I think it's better to drop the closing parenthesis to the following line, and align it with the beginning of the `$NAMES` variable.

File: apps/qubit/modules/settings/actions/identifierAction.class.php [hunk 2]
```diff
@@ -62,9 +63,13 @@ protected function addField($name)
...
       case 'inherit_code_informationobject':
       case 'inherit_code_dc_xml':
+      case 'prevent_duplicate_actor_identifiers':
         // Determine default value
         // (accession mask enabled setting doesn't get created in DB by default)
-        $defaults = array('accession_mask_enabled' => 1, 'inherit_code_dc_xml' => 0);
+        $defaults = array(
+          'accession_mask_enabled' => 1,
+          'inherit_code_dc_xml' => 0,
+          'prevent_duplicate_actor_identifiers' => 0);
```
Comment (djjuhasz) [inline apps/qubit/modules/settings/actions/identifierAction.class.php]: Cosmetic: closing parenthesis on it's own line

File: apps/qubit/modules/settings/templates/identifierSuccess.php [hunk 3]
```diff
@@ -64,6 +64,10 @@
           ->label(__('Inherit reference code (DC XML)'))
           ->renderRow() ?>
 
+        <?php echo $form->prevent_duplicate_actor_identifiers
```
Comment (djjuhasz) [inline apps/qubit/modules/settings/templates/identifierSuccess.php:67]: Code standard: Line is over 120 chars, I'd break like this:

```
        <?php echo $form->prevent_duplicate_actor_identifiers
          ->label(__(
              '%1% identifiers: prevent entry/import of duplicates', 
              array('%1%' => sfConfig::get('app_ui_label_actor'))
            ))
          ->renderRow() ?>
```

File: lib/task/import/csvAuthorityRecordImportTask.class.php [hunk 4]
```diff
@@ -234,11 +234,23 @@ public function execute($arguments = array(), $options = array())
       {
         if ($self->object)
         {
-          // Warn if identifier's already been used
+          // Warn or abort if identifier's already been used
           if (!empty($identifier = $self->columnValue('descriptionIdentifier'))
               && QubitValidatorActorDescriptionIdentifier::identifierUsedByAnotherActor($identifier, $self->object))
           {
-            print $self->logError(sprintf('Authority record identifier "%s" not unique.', $identifier));
+            $error = sfContext::getInstance()->i18n->__(
```
Comment (djjuhasz) [inline lib/task/import/csvAuthorityRecordImportTask.class.php]: Cosmetic: Indenting is unconventional here, I'd suggest:

```
            $error = sfContext::getInstance()
                      ->i18n
                      ->__(
                          '%1% identifier "%2%" not unique.',
                          array(
                            '%1%' => sfConfig::get('app_ui_label_actor'),
                            '%2%' => $identifier
                          )
                        );
```

**Comment by mcantelon** [comment]
@djjuhasz Thanks! I've updated the PR to address those issues.

File: lib/task/import/csvAuthorityRecordImportTask.class.php [hunk 5]
```diff
@@ -234,11 +234,29 @@ public function execute($arguments = array(), $options = array())
...
+                       );
+
+            if (sfConfig::get('app_prevent_duplicate_actor_identifiers', false))
+            {
+              $error .= sfContext::getInstance()->i18n->__(' Import aborted.');
+              throw new sfException($error);
+            }
+            else
+            {
+              print $self->logError($error);
```
Comment (djjuhasz) [inline lib/task/import/csvAuthorityRecordImportTask.class.php]: Cosmetic: should use `echo` instead of `print`

**Review by djjuhasz — APPROVED** [review]
One last minor change @mcantelon, then it's good to go :)

**Pull request merged**